python/cudf/cudf/core/frame.py

```diff
@@ -131,22 +135,34 @@ def _slice(self, arg):
                 """Step size is not supported other than None and 1"""
             )
 
+        # This is just to handle RangeIndex type, stop
+        # it from materializing unnecessarily
+        keep_index = True
+        if isinstance(self.index, RangeIndex):
+            keep_index = False
+
         if start < 0:
             start = start + num_rows
         if stop < 0:
             stop = stop + num_rows
 
         if start > stop:
-            return self._empty_like()
+            return self._empty_like(keep_index)
         else:
             start = len(self) if start > num_rows else start
             stop = len(self) if stop > num_rows else stop
 
             result = self._from_table(
-                libcudfxx.copying.table_slice(self, [start, stop])[0]
+                libcudfxx.copying.table_slice(self, [start, stop], keep_index)[
+                    0
+                ]
             )
 
-            result._copy_categories(self)
+            result._copy_categories(self, keep_index)
+            # Adding index of type RangeIndex back to
+            # result
+            if keep_index is False:
+                result.index = RangeIndex(start, stop)
```

Comment:
This assumes that `result` is a DataFrame or Series.

Additionally, this doesn't account for the initial state of the passed in `self.index` where say it was a `RangeIndex(5, 10)`, and I sliced `[1:3]`, the output should be `RangeIndex(6, 8)`.